CoreFoundation/URL.subproj/CFURLComponents.c

```diff
@@ -1409,6 +1409,9 @@ static Boolean _CFURLComponentsSetQueryItemsInternal(CFURLComponentsRef componen
                 if ( name && name != kCFNull ) {
                     if ( addPercentEncoding ) {
                         CFStringRef stringWithPercentEncoding = _CFStringCreateByAddingPercentEncodingWithAllowedCharacters(kCFAllocatorSystemDefault, name, queryNameValueAllowed);
+                        if ( !stringWithPercentEncoding ) {
```

Comment:
Why the change here vis-a-vis Darwin?

Reply:
I ported this condition from Darwin, following up internally.